apps/browser/qml/pages/components/SecureAction.qml

```diff
@@ -0,0 +1,33 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.0
+import com.jolla.settings.system 1.0
+import org.nemomobile.devicelock 1.0
+
+DeviceLockQuery {
```

Comment:
Looks that there's some weird flickering when this is fading away. Hmm, is there page (dialog) replace maybe happening?

Reply:
Yes, I get this too. I'm at a bit of a loss right now to understand why. Both cancel and accept use the same `pageStack.pop()` with `backNavigation` set to `false`, which should perform a fade back to the previous page. But the glitch when accepting is really obvious. Allowing back navigation solves the problem because it animates as a slide rather than a fade, but this doesn't seem to be a sensible approach (for multiple reasons).

 I'll persevere to try to find a solution.

Reply:
There's probably an absence of page background duct tape for that transition.

If you're not invested in the having the security prompt inline in the application*, then you can use the device lock API directly and it will display a system dialog instead which won't have any page stack issues. 

* I have thoughts on the security credentials of asking the user to type their security code into some app.

Reply:
Some privileged group misunderstanding aside, this appears to work.

```
Authenticator {
    id: deviceLockQuery
    property bool requireAuthentication: true
    property string message
    property var _resolve

    function perform(resolve) {
        if (requireAuthentication && availableMethods !== Authenticator.NoAuthentication) {
            _resolve = resolve

            deviceLockQuery.requestPermission(message, {})
        } else {
            resolve()
        }
    }

    onPermissionGranted: {
        if (_resolve) {
            requireAuthentication = false
            _resolve()
        }
    }
}
```

Reply:
Thanks Andrew, this has been really instructive. I've switched to using this code. If the browser is run without privileges, or there are no `availableMethods` for some other reason, then the password access methods are now disabled, which I think should have already been the case.

This also deals with the transition issues. Thanks again.

It's a separate issue, but the "full screen devicelock" approach is used in the WLAN editor (Settings > WLAN > Edit entry > unhide password). Perhaps that would also be better as an Authenticator rather than a DeviceLockQuery. It doesn't have the transition issues of course, but I feel like this "enter pin to show password" pattern is something we should be using in other places too, such as Accounts.

Reply:
Settings is a bit of a funny case because it has a number of settings pages that use inline security dialogs, including the actual device lock settings, and fingerprint settings. So do you have internal consistency in settings, or consistency in password fields, or flesh out the system dialogs so they can fully replace the inline ones?

Reply:
Yes, I can see the difficulty. I have my own opinions about how it should all work of course, but best left for another time. I'll resolve this conversation now, as I think we covered what's needed for this PR. Hopefully we can revisit the wider issue in the future.